have bound-with children get permanent location library for display

dnoneill opened the pull request:
closes #4349

cbeer commented:
Do we need to do all these work-arounds, or is it enough to initialize our bound-with items with the same data we use for regular items?

```
diff --git a/app/models/concerns/solr_holdings.rb b/app/models/concerns/solr_holdings.rb
index c535a78d..04c6b42c 100644
--- a/app/models/concerns/solr_holdings.rb
+++ b/app/models/concerns/solr_holdings.rb
@@ -57,10 +57,10 @@ module SolrHoldings
   end
 
   def bound_with_folio_items
-    @bound_with_folio_items ||= Array(holdings_json['holdings']).filter_map do |holdings_record|
-      next if holdings_record['boundWith'].blank?
+    @bound_with_folio_items ||= folio_holdings.filter_map do |holdings_record|
+      next unless holdings_record.bound_with_parent
 
-      Folio::Item.from_dynamic(holdings_record.dig('boundWith', 'item'))
+      Folio::Item.from_dynamic(holdings_record.bound_with_parent.item, holdings_record: holdings_record)
     end
   end
```

dnoneill commented:
@cbeer That works and is cleaner. I am closing this in favor of https://github.com/sul-dlss/SearchWorks/pull/4724. I implemented your suggestion and added tests to make sure it works correctly.
